packages/shared/src/graphql/bookmarks.ts

```diff
@@ -89,6 +109,45 @@ export const createBookmarkFolder = async ({
     .then((data) => data.createBookmarkList);
 };
 
+export const UPDATE_BOOKMARK_FOLDER = gql`
+  mutation UpdateBookmarkList($id: ID!, $name: String!, $icon: String) {
+    updateBookmarkList(id: $id, name: $name, icon: $icon) {
+      icon
+      name
+    }
+  }
+`;
+
+export const updateBookmarkFolder = async ({
+  id,
+  name,
+  icon,
+}: BookmarkFolder): Promise<BookmarkFolder> => {
+  return gqlClient
```

Comment:
Probably the reason why `onError` is not working as intended. We should probably use try-catch.